test/e2e/k8s_util.go

```diff
@@ -75,6 +77,8 @@ func (k *KubernetesUtils) GetPodsByLabel(ns string, key string, val string) ([]v
 		return p, nil
 	}
 
+	k.podLock.Lock()
+	defer k.podLock.Unlock()
```

Comment:
I mean L76 should be in the critical section, otherwise there would be be race condition between reading the map and writing the map.

Reply:
done